### DISCUSSION TITLE
MP3 stream sound interference

### DISCUSSION THREAD

**Discussion by skylai82** [schreibfaul1/ESP32-audioI2S#613] [Q&A] [answered]
Dear Team,

Stream this link https://cast.requestradio.in.th:830/stream?type=http&nocache=659
The sound was not smooth.
show the error below.
how to solve it?

21:02:14.881 -> info        syncword found at pos 0
21:02:14.881 -> info        MP3 decode error -2 : MAINDATA_UNDERFLOW
21:02:14.881 -> info        syncword found at pos 416
21:02:14.881 -> info        syncword found at pos 0
21:02:14.881 -> info        MP3 decode error -2 : MAINDATA_UNDERFLOW
21:02:14.881 -> info        syncword found at pos 225
21:02:14.914 -> info        syncword found at pos 0
21:02:14.914 -> info        MP3 decode error -9 : INVALID_HUFFCODES
21:02:14.914 -> info        syncword found at pos 190
21:02:14.914 -> info        syncword found at pos 0

**Comment by schreibfaul1** [comment]
Many audio streams can be played in the Arduino environment without any problems, but this depends on the size and order of the received data packets. Sometimes an optimisation is necessary. I have tested the stream  https://cast.requestradio.in.th:830/stream?type=http&nocache=659  with these [settings](https://github.com/schreibfaul1/ESP32-audioI2S/tree/master/examples/Better_WiFi_throughput)  and it works perfectly.

**Reply by skylai82** [reply]
https://cast.requestradio.in.th:830/stream?type=http&nocache=659  (MP3)
22:41:07.795 -> info        MP3 decode error -6 : INVALID_FRAMEHEADER
22:41:07.795 -> info        syncword found at pos 88
22:41:07.795 -> info        syncword found at pos 0
22:41:07.795 -> info        MP3 decode error -6 : INVALID_FRAMEHEADER
22:41:07.795 -> info        syncword found at pos 0
22:41:07.795 -> info        MP3 decode error -6 : INVALID_FRAMEHEADER
22:41:07.795 -> info        syncword found at pos 0
22:41:07.795 -> info        MP3 decode error -6 : INVALID_FRAMEHEADER
22:41:07.795 -> info        syncword found at pos 66
22:41:07.795 -> info        syncword found at pos 0
22:41:07.828 -> info        Channels: 2
22:41:07.828 -> info        SampleRate: 44100
22:41:07.828 -> info        BitsPerSample: 16
22:41:07.828 -> info        BitRate: 128000
22:41:08.625 -> info        slow stream, dropouts are possible

**ACC format no problem.**
2:47:02.513 -> info        Connect to new host: "https://23743.live.streamtheworld.com/UFM_1003AAC.aac"
22:47:02.546 -> info        PSRAM found, inputBufferSize: 638965 bytes
22:47:02.546 -> info        buffers freed, free Heap: 100704 bytes
22:47:04.208 -> info        SSL has been established in 1660 ms, free Heap: 54624 bytes
22:47:05.204 -> info        AACDecoder has been initialized, free Heap: 33996 bytes , free stack 4340 DWORDs
22:47:08.228 -> info        stream ready
22:47:08.228 -> info        syncword found at pos 0
22:47:08.228 -> info        Channels: 2
22:47:08.228 -> info        SampleRate: 22050
22:47:08.228 -> info        BitsPerSample: 16
22:47:08.228 -> info        BitRate: 59087
22:47:08.228 -> info        AAC HeaderFormat: ADTS
22:47:08.228 -> info        AAC Codec: MPEG-4 LowComplexity
22:47:08.294 -> info        VBR recognized, audioFileDuration is estimated

**Suspected**
1)May be source code MP3 decoder not support high BitRate: 128000 !!
2)ESP32-S2 MP3 stream not smooth.
3)MAX98357A DAC decode MP3 format not smooth.

**Comment by schreibfaul1** [comment, marked as answer]
You need a two-core µC. An ESP32-S2 is too weak.

**Reply by skylai82** [reply]
Thanks for your advice.  😘
